testsuite/MDAnalysisTests/analysis/test_gnm.py

```diff
@@ -37,119 +37,115 @@
     assert_almost_equal(time, np.arange(0, 1000, 100), decimal=4)
     assert_almost_equal(eigenvalues,
-      [ 2.0287113e-15, 4.1471575e-15, 1.8539533e-15, 4.3810359e-15,
-        3.9607304e-15, 4.1289113e-15, 2.5501084e-15, 4.0498182e-15,
-        4.2058769e-15, 3.9839431e-15])
+      [2.0287113e-15, 4.1471575e-15, 1.8539533e-15, 4.3810359e-15,
+       3.9607304e-15, 4.1289113e-15, 2.5501084e-15, 4.0498182e-15,
+       4.2058769e-15, 3.9839431e-15])
 
 
 def test_gnm_run_step(universe):
-    gnm = MDAnalysis.analysis.gnm.GNMAnalysis(universe)
+    gnm = mda.analysis.gnm.GNMAnalysis(universe)
     gnm.run(step=3)
     result = gnm.results
     assert len(result) == 4
     time, eigenvalues, eigenvectors = zip(*result)
     assert_almost_equal(time, np.arange(0, 1200, 300), decimal=4)
     assert_almost_equal(eigenvalues,
-      [ 2.0287113e-15, 4.3810359e-15, 2.5501084e-15, 3.9839431e-15])
+      [2.0287113e-15, 4.3810359e-15, 2.5501084e-15, 3.9839431e-15])
 
 
 def test_generate_kirchoff(universe):
-    gnm = MDAnalysis.analysis.gnm.GNMAnalysis(universe)
-    gnm.run()
+    gnm = mda.analysis.gnm.GNMAnalysis(universe)
     gen = gnm.generate_kirchoff()
     assert_almost_equal(gen[0],
-      [ 7,-1,-1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
-        0, 0, 0, 0, 0,-1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
-        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
-        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
-        0, 0, 0, 0, 0, 0,-1,-1,-1, 0, 0, 0, 0, 0, 0, 0, 0, 0,
-        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,-1, 0, 0, 0, 0,
-        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
-        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
-        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
-        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
-        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
-        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0])
+      [7,-1,-1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+       0, 0, 0, 0, 0,-1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+       0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+       0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+       0, 0, 0, 0, 0, 0,-1,-1,-1, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+       0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,-1, 0, 0, 0, 0,
+       0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+       0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+       0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+       0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+       0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+       0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0])
 
 
 def test_closeContactGNMAnalysis(universe):
-    gnm = MDAnalysis.analysis.gnm.closeContactGNMAnalysis(universe, weights="size")
-
-    gnm.run()
-
+    gnm = mda.analysis.gnm.closeContactGNMAnalysis(universe, weights="size")
+    gnm.run(stop=2)
```

Comment:
this is the speed up secret. Iterating over the whole trajectory just takes enormous amounts of time. Reducing to just looking if the first two frames are OK speeds things up significantly.